signer/src/transaction_coordinator.rs

```diff
@@ -1672,76 +1695,295 @@ where
+            // initial query only checks the total number of votes accepted by
+            // any signer).
+            if num_votes_accepted < params.signature_threshold {
+                tracing::warn!(
+                    request_id = req.request_id,
+                    num_votes_accepted,
+                    num_votes_rejected,
+                    num_votes_missing,
+                    required_votes = params.signature_threshold,
+                    reason = SKIP_REASON_INSUFFICIENT_VOTES,
+                    message = REQUEST_SKIPPED_MESSAGE
+                );
+                continue;
+            }
+
+            let withdrawal = utxo::WithdrawalRequest::from_model(req, votes);
+            eligible_withdrawals.push(withdrawal);
         }
 
-        let mut withdrawals = Vec::new();
+        Ok(eligible_withdrawals)
+    }
 
-        for req in pending_withdraw_requests {
+    /// TODO(#742): This function needs to filter deposit requests based on
+    /// time as well. We need to do this because deposit requests are locked
+    /// using OP_CSV, which lock up coins based on block height or
+    /// multiples of 512 seconds measure by the median time past.
+    #[tracing::instrument(skip_all)]
+    pub async fn get_eligible_pending_deposit_requests<DB>(
+        storage: &DB,
+        context_window: u16,
+        params: &GetPendingRequestsParams<'_>,
+    ) -> Result<Vec<utxo::DepositRequest>, Error>
+    where
+        DB: DbRead,
+    {
+        tracing::debug!("fetching eligible deposit requests");
+        let mut eligible_deposits: Vec<utxo::DepositRequest> = Vec::new();
+
+        // First, we fetch pending deposit requests with initial filtering
+        // done by the storage layer.
+        let pending_deposit_requests = storage
+            .get_pending_accepted_deposit_requests(
+                params.bitcoin_chain_tip.as_ref(),
+                context_window,
+                params.signature_threshold,
+            )
+            .await?;
+
+        // If there are no pending deposit requests, we can exit early.
+        if pending_deposit_requests.is_empty() {
+            tracing::debug!("no pending deposit requests eligible for consideration found");
+            return Ok(eligible_deposits);
+        }
+
+        // Iterate through each deposit request, fetch its _actual_ votes from
+        // storage and
+        for req in pending_deposit_requests {
             let votes = storage
-                .get_withdrawal_request_signer_votes(&req.qualified_id(), aggregate_key)
+                .get_deposit_request_signer_votes(&req.txid, req.output_index, params.aggregate_key)
                 .await?;
 
-            let withdrawal = utxo::WithdrawalRequest::from_model(req, votes);
-            withdrawals.push(withdrawal);
+            let deposit = utxo::DepositRequest::from_model(req, votes);
+            eligible_deposits.push(deposit);
         }
```

Comment:
This is interesting: we are not checking how many votes we have for the signers in the signing set identified by `aggregate_key` -- I guess we are relying on them being filtered out in `compute_optimal_packages`, but we could prefilter them already.
But even if we did, we are probably using the wrong `aggregate_key`, since we should check the signing set identified by `req.signers_public_key` instead.
We should probably address those points, unless I'm missing something. WDYT?

Reply:
mmm but the current signing set may not be able to (fully) sign for an old deposit if that was spent to a different aggregate key (and signing set); on the compliance level I think it would still be the same, as if you don't want to sign for an input you would just not sign for it (and we record the bitmap, so no foul play there).

Reply:
Opened https://github.com/stacks-network/sbtc/issues/1412 for the second point, something for another day.

Reply:
Alright, let's tackle that in a separate PR so we don't mess around with deposit validation logic here; it'll probably break some test somewhere too.

I guess you're creating a ticket for the 2nd point :)